sklearn/covariance/_shrunk_covariance.py

```diff
@@ -195,35 +218,27 @@ def fit(self, X, y=None):
 
 def ledoit_wolf_shrinkage(X, assume_centered=False, block_size=1000):
     """Estimate the shrunk Ledoit-Wolf covariance matrix.
-
     Read more in the :ref:`User Guide <shrunk_covariance>`.
-
     Parameters
     ----------
     X : array-like of shape (n_samples, n_features)
         Data from which to compute the Ledoit-Wolf shrunk covariance shrinkage.
-
     assume_centered : bool, default=False
         If True, data will not be centered before computation.
         Useful to work with data whose mean is significantly equal to
         zero but is not exactly zero.
         If False, data will be centered before computation.
-
     block_size : int, default=1000
         Size of blocks into which the covariance matrix will be split.
-
     Returns
     -------
     shrinkage : float
         Coefficient in the convex combination used for the computation
         of the shrunk estimate.
-
     Notes
     -----
     The regularized (shrunk) covariance is:
-
     (1 - shrinkage) * cov + shrinkage * mu * np.identity(n_features)
-
     where mu = trace(cov) / n_features
     """
     X = check_array(X)
```

Comment:
I see what you mean. We again check and raise a new warning. If we deprecate and make this class private then, we can avoid making these checks because we don't have any source code that can reach these warning.

Reply:
So what is the deprecation policy here? This function is technically public, but it is not mentioned in the [docs](https://scikit-learn.org/dev/modules/classes.html#module-sklearn.covariance)

Reply:
We can do the deprecation in another PR. The idea will be

- create a private function with the same signature
- make the public function calling the private one
- if people call the private one, they should get a warning that it is deprecated in the current version (e.g. 1.2) and it will be removed in 2 minor versions (1.4).

You can find more details regarding the deprecation handling in scikit-learn here: https://scikit-learn.org/dev/developers/contributing.html#deprecation

It is true that we never documented but we tend to be extra careful to remove something that was half public.